pkg/operator/k8sutil/node.go

```diff
@@ -34,27 +34,27 @@ import (
 // ValidNodeNoSched returns true if the node (1) meets Rook's placement terms,
 // and (2) is ready. Unlike ValidNode, this method will ignore the
 // Node.Spec.Unschedulable flag. False otherwise.
-func ValidNodeNoSched(node v1.Node, placement cephv1.Placement) (bool, error) {
+func ValidNodeNoSched(node v1.Node, placement cephv1.Placement) (bool, string, error) {
 	p, err := NodeMeetsPlacementTerms(node, placement, false)
 	if err != nil {
-		return false, fmt.Errorf("failed to check if node meets Rook placement terms. %+v", err)
+		return false, "", fmt.Errorf("failed to check if node meets Rook placement terms. %+v", err)
 	}
 	if !p {
-		return false, nil
+		return false, "placement settings do not match", nil
 	}
 
 	if !NodeIsReady(node) {
-		return false, nil
+		return false, "node is not ready", nil
 	}
 
-	return true, nil
+	return true, "", nil
 }
 
 // ValidNode returns true if the node (1) is schedulable, (2) meets Rook's placement terms, and
 // (3) is ready. False otherwise.
-func ValidNode(node v1.Node, placement cephv1.Placement) (bool, error) {
+func ValidNode(node v1.Node, placement cephv1.Placement) (bool, string, error) {
```

Comment:
Can't we just return an `error` that is populated appropriately instead of the current combination? If `err != nil`, we just error out and log? I hope this would simplify the function's signature.

Reply:
There are error conditions that should fail the reconcile, and there are these messages that we want to write to the log without failing the reconcile. I didn't really love the complex func signature either, but this seemed like the best way.

Reply:
What if we use `errors.Is()` to log the errors we know we don't want the reconcile to fail? It's a matter of registering `errors.New("node is not ready")` +  `errors.Is()` to skip when desired.